src/collectors/simple/simple.py

```diff
@@ -0,0 +1,203 @@
+            file.close()
+
+            for line in lines:
+                if line.startswith('cpu '):
+                    elements = line.split()
+                    self.collect_cpu_proc(elements)
+        else:
+            total_time = psutil.cpu_times()
+            self.collect_cpu_psutil(total_time)
+
+        # Memory collection only exists for /proc/meminfo
+        if os.access(self.PROC_MEM, os.R_OK):
+            file = open(self.PROC_MEM)
+            lines = file.read().splitlines()
+            file.close()
+
+            self.collect_memory_proc(lines)
+
+        if os.access(self.PROC_DISKSTATS, os.R_OK):
+            file = open(self.PROC_DISKSTATS)
+            lines = file.read().splitlines()
+            file.close()
+
+            self.collect_disk_stats_proc(lines)
+        else:
+            disks = psutil.disk_io_counters(True)
+
+            self.collect_disk_stats_psutil(disks)
+
+        if os.access(self.PROC_DISKSPACE, os.R_OK):
+            file = open(self.PROC_DISKSPACE)
+            lines = file.read().splitlines()
+            file.close()
+
+            # Filter to collectable mount points
+            mount_points = [line.split()[1] for line in lines if self.is_disk_collectable(line.split()[2], line.split()[1], line.split()[0])]
+            self.collect_disk_space_proc(mount_points)
+        else:
+            partitions = psutil.disk_partitions(False)
+            mount_points = map(lambda partition: partition.mountpoint, partitions)
+            self.collect_disk_space_proc(mount_points)
+
+        return True
+
+    def collect_cpu_proc(self, elements):
+        # Compute all CPU usage values from /proc/stat counter values
+        user = self.derivative('cpu.total.user', long(elements[1]), diamond.collector.MAX_COUNTER)
+        nice = self.derivative('cpu.total.nice', long(elements[2]), diamond.collector.MAX_COUNTER)
+        system = self.derivative('cpu.total.system', long(elements[3]), diamond.collector.MAX_COUNTER)
+        idle = self.derivative('cpu.total.idle', long(elements[4]), diamond.collector.MAX_COUNTER)
+        iowait = self.derivative('cpu.total.iowait', long(elements[5]), diamond.collector.MAX_COUNTER)
+        irq = self.derivative('cpu.total.irq', long(elements[6]), diamond.collector.MAX_COUNTER)
+        softirq = self.derivative('cpu.total.softirq', long(elements[7]), diamond.collector.MAX_COUNTER)
+        steal = self.derivative('cpu.total.steal', long(elements[8]), diamond.collector.MAX_COUNTER)
+        guest = self.derivative('cpu.total.guest', long(elements[9]), diamond.collector.MAX_COUNTER)
+        guest_nice = self.derivative('cpu.total.guest_nice', long(elements[10]), diamond.collector.MAX_COUNTER)
+
+        total = sum([user, nice, system, idle, iowait, irq, softirq, steal, guest, guest_nice])
+
+        # Derivatives take one cycle to warm up
+        if total != 0:
+            self.publish('cpu.total.utilization.percent', (total - idle) / total * 100)
```

Comment:
Do we have to keep the full tree for this metric? 
It seems odd to have 3 levels for this metric but only 1 for the others.

Reply:
This is to replicate https://github.com/netuitive-community-packages/netuitive-packages-linux/blob/4393a315e6261284fd3985c182ccec9be6694cb5/analyticConfigurations/computed_metric-linux.json#L24, so yes.